### PR TITLE
Chore: Convert to typescript the open slash commands files

### PR DESCRIPTION
**Pull request by eduardofcabrera**
## Proposed changes (including videos or screenshots)

Convert to typescript the open slash commands files

### CODE REVIEW

File: app/slashcommands-open/client/client.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	let room = params.trim();
+	const typeStr = room[0];
+	let type: string[] | null;
+	if (typeStr === '#') {
+		type = ['c', 'p'];
+	} else if (typeStr === '@') {
+		type = ['d'];
+	} else {
+		type = null;
+	}
```
Comment (ostjen) [inline app/slashcommands-open/client/client.ts:18]: why did you removed the dict?

Comment (ostjen) [inline app/slashcommands-open/client/client.ts:18]: why did you removed the dict?

Comment (ostjen) [inline app/slashcommands-open/client/client.ts:18]: why did you removed the dict?

Comment (ostjen) [inline app/slashcommands-open/client/client.ts:18]: why did you remove the dict?

**Review by ostjen — CHANGES_REQUESTED** [review]

File: app/slashcommands-open/client/client.ts (hunk 1 above)
Comment (eduardofcabrera) [reply app/slashcommands-open/client/client.ts:18]: Because it was giving type erros, so I would need to create an interface to the dict. I prefered to just create an if-else to the cases.

Comment (eduardofcabrera) [reply app/slashcommands-open/client/client.ts:18]: It was giving that string can't be used to index the dict.

File: app/slashcommands-open/client/client.ts [hunk 2]
```diff
@@ -1,26 +1,22 @@
...
-		return;
-	}
-
 	let room = params.trim();
 	const type = dict[room[0]];
 	room = room.replace(/#|@/, '');
-
 	const query = {
 		name: room,
+		t: {},
```
[the remark this answers is not in the document]
Comment (eduardofcabrera) [reply app/slashcommands-open/client/client.ts:19]: on line 27, requires the t property of the query, and the typescript was complaining about typing, so I just set it to update later.

[the remark this answers is not in the document]
Comment (ostjen) [reply app/slashcommands-open/client/client.ts:19]: ok, but wouldn't be better to do the line 27 assignment here, instead of initializing with an empty object and reassigning after?

**Review by ostjen — CHANGES_REQUESTED** [review]

File: app/slashcommands-open/client/client.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import { Subscriptions, ChatSubscription } from '../../models/client';
+
+function Open(_command: 'open', params: string, _item: Record<string, string>): void {
+	const dict: Record<string, string[]> = {
+		'#': ['c', 'p'],
+		'@': ['d'],
+	};
+
+	let room = params.trim();
+	const type = dict[room[0]];
```
Comment (ostjen) [inline app/slashcommands-open/client/client.ts:15]: since you removed the check for type existing it'd be better to
`const type = dict[room[0]] || {};`

**Review by ggazzo — APPROVED** [review]
